src/plugins/drawing/model/drawing-content.ts

```diff
@@ -72,9 +72,13 @@ export const DrawingContentModel = TileContentModel
       return self.objects.map(object => ({
         objectId: object.id,
         objectType: object.type,
-        tileId 
+        tileId
       }));
     },
+    // Note: We could be using MST's resolveIdentifier instead of making our own
+    // objectMap. This would be more efficient, however this will really
+    // enforce unique ids for all drawing objects regardless of which tile they
+    // are in.
```

Comment:
`resolveIdentifier()` can be used to get from an id to a model object, but the tile will still be involved:
- the tile determines which ids are linkable (not everything with an id is necessarily linkable)
- not all ids are MST identifiers, for instance table cells are likely to use constructed ids
- the tile has to provide boundary information for each object
so I think leaving it the responsibility of the individual tiles makes sense.

Reply:
This objectMap existed and was used before the sparrows work was added. So my suggestion here is to only update the internal drawing tile code with resolveIdentifier. Specifically the change would be to find all of the places in `plugins/drawing/` that refer to `content.objectMap[id]` and replace them with `resolveIdentifier(DrawingObject, content, id)`.  In theory resolveIdentifier should find objects that are "subtypes" of DrawingObject at least the MST code I was working in with the identifierCache looked like it would handle that.

As I wrote that I realized the resolveIdentifier could make sure the object is a child of the object passed to resolveIdentifier.  So then if a drawing object id was re-used in two tiles it wouldn't be a problem. However I looked, at the code and resolveIdentifier finds the tree root of the passed in object and doesn't do any path comparison. In other words if the same id was used it would cause problems. It would also cause problems if the same id was used by two different types of drawing objects. The references would be stored correctly in the identifierCache, but the lookup would only find one of the objects since the "parent" type of DrawingObject was used instead of the object's specific type.